codec/types/any.go

```diff
@@ -68,11 +68,29 @@ func NewAnyWithValue(value proto.Message) (*Any, error) {
 	return any, nil
 }
 
-// Pack packs the value x in the Any or returns an error. This also caches
-// the packed value so that it can be retrieved from GetCachedValue without
-// unmarshaling
-func (any *Any) Pack(x proto.Message) error {
-	any.TypeUrl = "/" + proto.MessageName(x)
+// NewAnyWithTypeURL constructs a new Any packed with the value provided
+// and a custom TypeURL. It returns an error if that value couldn't be packed.
+// This also cachesthe packed value so that it can be retrieved from
+// GetCachedValue without unmarshaling.
+//
+// Ex:
+// This will allow us to pack service methods in Any's using the full method name
+// as the type URL and the request body as the value.
+func NewAnyWithTypeURL(typeURL string, value proto.Message) (*Any, error) {
```

Comment:
This function looks dangerous. 

Is the protobuf generated code already providing it? Maybe there is a method already for this?

Reply:
> Is the protobuf generated code already providing it? Maybe there is a method already for this?

That wouldn't work, because we have our own implementation of Any.